plugins/modules/ndo_tenant_igmp_interface_policy.py

```diff
@@ -0,0 +1,534 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_tenant_igmp_interface_policy
+short_description: Manage Internet Group Management Protocol (IGMP) Interface Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IGMP Interface Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the tenant template.
+    type: str
+    aliases: [ tenant_template ]
+    required: true
+  template_id:
+    description:
+    - The ID of the L3Out template.
```

Comment:
- This parameter or O(template) is required.

Reply:
changed